app/lib/actions/remote_execution/run_host_job.rb

```diff
@@ -55,6 +55,10 @@ def find_template_invocation(job_invocation, host)
       end
 
       def find_ip_or_hostname(host)
+        if host.execution_interface && host.execution_interface.ip.present?
+          return host.execution_interface.ip
+        end
+
```

Comment:
would it make sense to prefer `provision` and then `primary` interface before we go through all? sorry, I know it's not strictly related to your change